homeassistant/components/zwave_js/__init__.py

```diff
@@ -25,7 +25,7 @@
 from .websocket_api import async_register_api
 
 LOGGER = logging.getLogger(__name__)
-CONNECT_TIMEOUT = 10
+CONNECT_TIMEOUT = 300
```

Comment:
Does it take 5 min to send all the data or is this time increased to wait for the zwave devices to report to the controller?

Reply:
the init call can really take some time. I noticed the same on my main mesh.
The entire integration setup now depends on the initialize (first full state received) while it should only wait for the connect call. See discussion on discord. Best take would be to split the two up and in the config flow setup only wait for a successfull connect and not for the initialize to account for slow networks, large meshes etc.

Reply:
Ok, so it's still just about the time it takes to send/receive data over the websocket, not about waiting for devices to report.

Reply:
correct

Reply:
Wait. We send the state right away. Is it really that long? Do we have timestamps of how fast connect and state is received? Can we add some debugging

Reply:
on my main setup it takes about 8 seconds only to receive the full state (raw websocket message), let alone some json parsing that needs to be done in the library. And this is with fast hardware.

Reply:
Other solution to tackle this without increasing timeouts:
https://github.com/home-assistant/core/pull/45495

Reply:
see above comment's init-log.txt to see how long it is taking for the response to arrive (about 80 seconds, if the python-zwave-server log is to be believed). This is not fast hardware, but it isn't the slowest out there either...